services/supervisor/tests/supervisor/config.rs

```diff
@@ -0,0 +1,844 @@
+
+    testkit.create_block_with_transaction(sign_config_propose_transaction(
+        &testkit,
+        propose,
+        initiator_id,
+    ));
+    let signed_txs = build_confirmation_transactions(&testkit, proposal_hash, initiator_id);
+    testkit.create_block_with_transactions(signed_txs);
+
+    testkit.create_blocks_until(CFG_CHANGE_HEIGHT);
+
+    assert_eq!(config_propose_entry(&testkit), None);
+    check_service_actual_param(&testkit, Some(params));
+}
+
+#[test]
+fn test_discard_errored_service_config_change() {
+    let mut testkit = testkit_with_change_service_and_static_instance(4);
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+
+    let params = "I am a discarded parameter".to_owned();
+    let new_consensus_config = consensus_config_propose_first_variant(&testkit);
+
+    let propose = ConfigProposeConstructor::new(CFG_CHANGE_HEIGHT)
+        .extend_service_config_propose(params.clone())
+        .extend_service_config_propose("error".to_string())
+        .extend_consensus_config_propose(new_consensus_config)
+        .config_propose();
+
+    let proposal_hash = propose.object_hash();
+
+    testkit.create_block_with_transaction(sign_config_propose_transaction(
+        &testkit,
+        propose,
+        initiator_id,
+    ));
+    let signed_txs = build_confirmation_transactions(&testkit, proposal_hash, initiator_id);
+    testkit.create_block_with_transactions(signed_txs);
+    testkit.create_blocks_until(CFG_CHANGE_HEIGHT);
+
+    assert_eq!(config_propose_entry(&testkit), None);
+
+    check_service_actual_param(&testkit, None);
+    assert_eq!(testkit.network().us().validator_id(), Some(initiator_id));
+}
+
+#[test]
+fn test_discard_panicked_service_config_change() {
+    let mut testkit = testkit_with_change_service_and_static_instance(4);
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+
+    let params = "I am a discarded parameter".to_owned();
+    let new_consensus_config = consensus_config_propose_first_variant(&testkit);
+
+    let propose = ConfigProposeConstructor::new(CFG_CHANGE_HEIGHT)
+        .extend_service_config_propose(params.clone())
+        .extend_service_config_propose("panic".to_string())
+        .extend_consensus_config_propose(new_consensus_config)
+        .config_propose();
+
+    let proposal_hash = propose.object_hash();
+
+    testkit.create_block_with_transaction(sign_config_propose_transaction(
+        &testkit,
+        propose,
+        initiator_id,
+    ));
+    let signed_txs = build_confirmation_transactions(&testkit, proposal_hash, initiator_id);
+    testkit.create_block_with_transactions(signed_txs);
+    testkit.create_blocks_until(CFG_CHANGE_HEIGHT);
+
+    assert_eq!(config_propose_entry(&testkit), None);
+
+    check_service_actual_param(&testkit, None);
+    assert_eq!(testkit.network().us().validator_id(), Some(initiator_id));
+}
+
+#[test]
+fn test_incorrect_actual_from_field() {
```

Comment:
Are this test same as test_create_expired_proposal?

Reply:
Yes. I will remove test_create_expired_proposal.